### PR TITLE
feat: implement blocked channels

### PR DESCRIPTION
**Pull request by johnletey**
This PR implements a blocklist approach to IBC transfers.

A new E2E test has been added to demonstrate this feature:

- Transferring out of Noble on unblocked channel. ✅ 
- Transferring back to Noble on unblocked channel. ✅
- Transferring out of Noble on blocked channel. ❌
- Transferring back to Noble on blocked channel. ✅

## Summary by CodeRabbit

- **New Features**
  - Introduced commands to add and remove blocked channels.
  - Added functionality for querying blocked channels.
  - Events `BlockedChannelAdded` and `BlockedChannelRemoved` now emitted on relevant transactions.
  
- **Bug Fixes**
  - Updated command descriptions for consistency.

- **Tests**
  - Added test cases for adding and removing blocked channels.
  - Added test case for querying blocked channels.
  - Enhanced transfer restriction logic with new test for blocked channels.

- **Documentation**
  - Updated documentation to include new message types and events related to blocked channels.

### CODE REVIEW

File: x/aura/types/msgs.go [hunk 1]
```diff
@@ -400,3 +401,61 @@ func (msg *MsgRemovePauser) GetSignBytes() []byte {
...
+var _ legacytx.LegacyMsg = &MsgRemoveBlockedChannel{}
+
+func (msg *MsgRemoveBlockedChannel) ValidateBasic() error {
+	if _, err := sdk.AccAddressFromBech32(msg.Signer); err != nil {
+		return fmt.Errorf("invalid signer address (%s): %w", msg.Signer, err)
+	}
+
+	if !channeltypes.IsValidChannelID(msg.Channel) {
+		return fmt.Errorf("invalid channel (%s)", msg.Channel)
+	}
```
Comment (boojamya) [inline x/aura/types/msgs.go:445]: This check probably isn't necessary since it is impossible to add an invalid channel. But I guess it doesn't hurt.

**Review by boojamya — APPROVED** [review]
Very clean. Nice job!

**Review by g-luca — APPROVED** [review]
lgtm!

File: e2e/main_test.go [hunk 2]
```diff
@@ -191,3 +193,93 @@ func TestMintTransferBlockBurn(t *testing.T) {
...
+	// ACT: Attempt to transfer out of Noble, channel is blocked.
+	_, err = wrapper.chain.SendIBCTransfer(ctx, "channel-0", wrapper.alice.KeyName(), ibc.WalletAmount{
+		Address: wrapper.charlie.FormattedAddress(),
+		Denom:   "ausdy",
+		Amount:  ONE.Int64(),
+	}, ibc.TransferOptions{})
+	// ASSERT: The action should've failed.
+	require.ErrorContains(t, err, "ausdy transfers are blocked on channel-0")
+
+	// ACT: Attempt to transfer back to Noble, channel is blocked.
```
Comment (fafrd) [inline e2e/main_test.go:270]:
```suggestion
	// ACT: Attempt to transfer back to Noble, channel is allowed.
```

**Review by fafrd — COMMENTED** [review]
Looks great, 1 small suggestion for an incorrect code comment. ty for making an e2e test!

**Review by fafrd — APPROVED** [review]

**Pull request merged**